### PR TITLE
Site endpoint: Add is_commercial to options

### LINKED ISSUE #32617: Paid Stats: create a flag to identify whether a site is commercial or not

### PR DESCRIPTION
**Pull request by kangzj**
Fixes: #32617 

## Proposed changes:

Added `is_commercial` property to /sites/{siteId} endpoint (inside options attribute), so Calypso can get whether a site is commercial or not and determine whether to allow the purchase of Free / Personal products.

The option is filled on wpcom: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Swrgcnpx%2Qfvgr%2Qcebsvyrf%2Spynff.wrgcnpx%2Qfvgr%2Qcebsvyrf%2Qpbzzrepvny%2Qfgngr.cuc%3Se%3Q31qn5255%23133-og

The value of `is_commercial` has 3 possibilities:
- `true`: commercial
- `false`: non-commercial
- `null`: to be determined

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Apply this PR change in your sandbox with the command: `bin/jetpack-downloader test jetpack add/is-commercial-to-options`
* Sandbox `public-api.wordpress.com`
* Open `https://developer.wordpress.com/docs/api/console/`, access the `/sites/{siteId}` wpcom v1.1
* Ensure you see `is_commercial` in the `options` object
* Please test with Jetpack self hosted, Atomic and Simple sites
* Check `/me/sites` as well

### CODE REVIEW

**Review by grzegorz-cp — DISMISSED** [review]
I tested with a JN website and my simple site. `/me/sites` is fine as well.

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 1]
```diff
@@ -271,6 +272,7 @@ class WPCOM_JSON_API_GET_Site_Endpoint extends WPCOM_JSON_API_Endpoint {
 		'blogging_prompts_settings',
 		'wpcom_production_blog_id',
 		'wpcom_staging_blog_ids',
+		'is_commercial',
```
Comment (kangzj) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-get-site-endpoint.php:275]: The option is stored in WPCOM, so we need it to be added / overridden by WPCOM here.

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 2]
```diff
@@ -1512,4 +1512,14 @@ public function can_blaze() {
 	public function get_wpcom_site_setup() {
 		return get_option( 'wpcom_site_setup' );
 	}
+
+	/**
+	 * Return true if the site is a commercial site. If not determined yet, it returns null.
```
Comment (sotirispl) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]: To clarify that we have 3 different states for `is_commercial`.
- True = the site is commercial
- False = the site is not commercial
- NULL = the commercial status is not yet defined

I tested the code of this function, and it gets all of these states correctly, but the PHPDoc should be updated.

Comment (kangzj) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: Thanks for the review @sotirispl 👍 

Updated PHPDoc in 0979f04f10.

**Review by grzegorz-cp — APPROVED** [review]
👍

**Pull request merged**